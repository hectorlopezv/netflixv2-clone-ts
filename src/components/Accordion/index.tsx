import React, {useState, useContext, createContext} from 'react'
import {Container, Title, Frame, Item, Header, Body} from './styles/Accordion';


const ToogleContext = createContext<{ToogleShow:any, setToogleShow:any}>({ToogleShow: false, setToogleShow: () => {}});
export interface AccordionProps {
    
}
 
const Accordion: React.FC<AccordionProps> = ({children, ...props}) => {
    return ( 
        <Container {...props}>
            {children}
        </Container>
     );
}
 
export default Accordion;

export interface Accordion_TitleProps {
    
}
 
export const Accordion_Title: React.FC<Accordion_TitleProps> = ({children, ...props}) => {
    return (  
        <Title {...props}>{children}</Title>
    );
}

export interface Accordion_ItemProps {
    
}


export interface ItemProps {//primary Container
    
}
 
export const Accordion_Item: React.FC<ItemProps> = ({children, ...props}) => {
    
    const [ToogleShow, setToogleShow] = useState(false);
 
    return (  
        <ToogleContext.Provider value={{ToogleShow, setToogleShow}}>
            <Item {...props}>{children}</Item>
        </ToogleContext.Provider>
    );
}


export const Accordion_Frame: React.FC<Accordion_ItemProps> = ({children, ...props}) => {
    return ( 
        <Frame {...props}>{children}</Frame>
     );
}

export interface Accordion_HeaderProps {
    
}
 
export const Accordion_Header: React.FC<Accordion_HeaderProps> = ({children, ...props}) => {
    const {setToogleShow} = useContext<{ToogleShow:any, setToogleShow: any}>(ToogleContext);
    
    return (  
        <Header onClick={() => setToogleShow((latest: boolean) => !latest)} {...props}>{children}</Header>
    );
}

export interface Accordion_BodyProps {
    
}
 
export const Accordion_Body: React.FC<Accordion_BodyProps> = ({children, ...props}) => {
    
    const {ToogleShow} = useContext<{ToogleShow:any, setToogleShow: any}>(ToogleContext);

    return ToogleShow? <Body {...props}>{children}</Body> : null;

}
 

 


 


 

