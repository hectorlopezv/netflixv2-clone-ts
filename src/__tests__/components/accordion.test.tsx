import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import faqsData from '../../fixtures/faqs.json';
import Accordion, {
    Accordion_Title, 
    Accordion_Item, 
    Accordion_Frame, 
    Accordion_Header, 
    Accordion_Body  
} from '../../components/Accordion';

describe('Testing Accordion', () => {

    it('test <Accordion /> with populated data', () => {

        const {container, getByText} = render(
            <Accordion>
            <Accordion_Title>Frequently Asked Questions</Accordion_Title>
            {faqsData.map((item: any) => {
                return <Accordion_Item key={item.id}>
                    <Accordion_Header>{item.header}</Accordion_Header>
                    <Accordion_Body>{item.body}</Accordion_Body>
                </Accordion_Item>
            })}
        </Accordion>
        );

        expect(getByText('What is Netflix?')).toBeTruthy();
        expect(getByText('How much does Netflix cost?')).toBeTruthy();
        expect(getByText('Where can I watch?')).toBeTruthy();
        expect(getByText('How do I cancel?')).toBeTruthy();
        expect(getByText('What can I watch on Netflix?')).toBeTruthy();
    });

    it('opens and closes the accordion <Accordion /> Component', () => {
        const {container, getByText, queryByText} = render(
            <Accordion>
            <Accordion_Title>Frequently Asked Questions</Accordion_Title>
            {faqsData.map((item: any) => {
                return <Accordion_Item key={item.id}>
                    <Accordion_Header>{item.header}</Accordion_Header>
                    <Accordion_Body>{item.body}</Accordion_Body>
                </Accordion_Item>
            })}
        </Accordion>
        );
        const text = "Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices.\n\nYou can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!";
        
        expect(queryByText(text)).toBeFalsy();

        // //we expect now to be true when we click it
        // fireEvent.click(queryByText("What is Netflix?")!);
        // expect(queryByText(text)).toBeTruthy();

        // fireEvent.click(queryByText("What is Netflix?")!);
        // expect(queryByText(text)).toBeFalsy();
        expect(container.firstChild).toMatchSnapshot();
    
    });
});