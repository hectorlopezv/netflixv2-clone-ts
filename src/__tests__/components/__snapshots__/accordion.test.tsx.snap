// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Accordion opens and closes the accordion <Accordion /> Component 1`] = `
<div
  class="Accordion__Container-sc-1llfk3v-0 ipGnGp"
>
  <div
    class="Accordion__Inner-sc-1llfk3v-1 eLDdmq"
  >
    <h1
      class="Accordion__Title-sc-1llfk3v-6 fiTCHn"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="Accordion__Item-sc-1llfk3v-2 iDbBps"
    >
      <div
        class="Accordion__Header-sc-1llfk3v-3 dLdwhC"
      >
        What is Netflix?
        <img
          alt="Close"
          src="images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="Accordion__Item-sc-1llfk3v-2 iDbBps"
    >
      <div
        class="Accordion__Header-sc-1llfk3v-3 dLdwhC"
      >
        How much does Netflix cost?
        <img
          alt="Close"
          src="images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="Accordion__Item-sc-1llfk3v-2 iDbBps"
    >
      <div
        class="Accordion__Header-sc-1llfk3v-3 dLdwhC"
      >
        Where can I watch?
        <img
          alt="Close"
          src="images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="Accordion__Item-sc-1llfk3v-2 iDbBps"
    >
      <div
        class="Accordion__Header-sc-1llfk3v-3 dLdwhC"
      >
        How do I cancel?
        <img
          alt="Close"
          src="images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="Accordion__Item-sc-1llfk3v-2 iDbBps"
    >
      <div
        class="Accordion__Header-sc-1llfk3v-3 dLdwhC"
      >
        What can I watch on Netflix?
        <img
          alt="Close"
          src="images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;
